src/vmm/src/vstate/vcpu/aarch64.rs

```diff
@@ -86,14 +87,18 @@ impl KvmVcpu {
     /// # Arguments
     ///
     /// * `guest_mem` - The guest memory used by this microvm.
-    /// * `kernel_load_addr` - Offset from `guest_mem` at which the kernel is loaded.
+    /// * `kernel_entry_point` - Specifies the boot protocol and offset from `guest_mem` at which
+    ///   the kernel starts.
     /// * `vcpu_config` - The vCPU configuration.
     pub fn configure(
         &mut self,
         guest_mem: &GuestMemoryMmap,
-        kernel_load_addr: GuestAddress,
+        kernel_entry_point: EntryPoint,
         vcpu_config: &VcpuConfig,
     ) -> Result<(), KvmVcpuError> {
+        // aarch64 only has LinuxBoot
+        assert_eq!(kernel_entry_point.protocol, BootProtocol::LinuxBoot);
```

Comment:
Usually we go for `debug_assert` only. However, in this case it'd be nice if we could statically guarantee that `BootProtocol::PvhBoot` cannot reach the `.configure` function on aarch64, because if that is ever possible then something is fundamentally broken somewhere. Maybe just putting the `PvhBoot` variant behind a `#[cfg(x86)]` would be a solution here to prevent anything like that from ever happening.